### DISCUSSION TITLE
UPS Property Rate.TotalCharges returns one hundred times the actual cost

### DISCUSSION THREAD

**Discussion by ebaj-tech** [alexeybusygin/ShippingRates#71] [Q&A] [answered]
When I read the shipping costs after calling GetRatesAsync(), I get a hundred times the actual cost. It is noticeable that there are no cent amounts. Is this intentional? Are the shipping costs returned in cent amounts by UPS?

**Comment by alexeybusygin** [comment]
UPS returns shipping costs with cents. Here is the output from the Sample app:

```
UPS
        01
        UPS Next Day Air
        292.63 USD
        8/13/2024 10:30:00 AM
UPS
        02
        UPS Second Day Air
        153.01 USD
        8/14/2024 11:59:00 PM
UPS
        03
        UPS Ground
        57.24 USD
        12/31/9999 11:59:00 PM
```

Could you share what you get in the response?

**Reply by ebaj-tech** [reply]
UPS
        07
        UPS Worldwide Express
        4403 EUR
        31.12.9999 23:59:00
UPS
        11
        UPS Standard
        2565 EUR
        31.12.9999 23:59:00
UPS
        54
        UPS Worldwide Express Plus
        7765 EUR
        31.12.9999 23:59:00
UPS
        65
        UPS Express Saver
        3010 EUR
        31.12.9999 23:59:00

**Reply by ebaj-tech** [reply]
Looking for that behavior, I found this line of code in **UPSProvider.cs**, **ParseResponse()**:
var totalCharges = Convert.ToDecimal(rate.TotalCharges.MonetaryValue);
Should this conversion use the current culture settings?

**Reply by alexeybusygin** [reply, marked as answer]
You were right, and the problem was here! Could you please check v2.2.1 and see if the issue is solved for you?

**Reply by ebaj-tech** [reply]
Perfekt, now it works fine. Thank You!
